platform/mbed_mem_trace.cpp

```diff
@@ -39,42 +45,50 @@ void mbed_mem_trace_set_callback(mbed_mem_trace_cb_t cb) {
     mem_trace_cb = cb;
 }
 
+void mbed_mem_trace_lock()
+{
+    mem_trace_mutex->lock();
+    core_util_atomic_incr_u8(&trace_lock_count, 1);
```

Comment:
nitpick - this doesn't need to be atomically incremented since this is protected by the mutex.

Reply:
Updated